cmd/commands/connection/util.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2020 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package connection
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/mysteriumnetwork/node/datasize"
+	"github.com/mysteriumnetwork/node/money"
+	"github.com/mysteriumnetwork/node/tequilapi/contract"
+)
+
+func printProposal(p *contract.ProposalDTO) {
+	ppm, ok := pricePerMinute(p.PaymentMethod)
+	if !ok {
+		return
+	}
+
+	ppgb, ok := pricePerGB(p.PaymentMethod)
+	if !ok {
+		return
+	}
+
+	fmt.Printf("| Identity: %s | Type: %s | Country: %s | Price: %s/min   %s/GB |\n",
+		p.ProviderID,
+		p.ServiceDefinition.LocationOriginate.NodeType,
+		p.ServiceDefinition.LocationOriginate.Country,
+		ppm,
+		ppgb,
+	)
+}
+
+func pricePerMinute(pm contract.PaymentMethodDTO) (string, bool) {
+	s := time.Duration(pm.Rate.PerSeconds) * time.Second
+	min := new(big.Float).SetFloat64(s.Minutes())
+	if min.Cmp(big.NewFloat(0)) == 0 {
+		// If a node is missconfigured and somehow reports Rate as 0
```

Comment:
Some nodes can provide a service for free, does this prevent user from seen such nodes?